terraform/batch.tf

```diff
@@ -1,10 +1,11 @@
 module "aws_batch_compute" {
   source = "./batch_compute"
-
+  name = "default"
   subnets            = "${join(",", formatlist("\"%s\"", module.vpc_batch.subnets))}"
   key_name           = "${var.key_name}"
   admin_cidr_ingress = "${var.admin_cidr_ingress}"
   vpc_id             = "${module.vpc_batch.vpc_id}"
+  image_id = "ami-0657a97f"
```

Comment:
Probably possible to generate a tfvars file, but you might be able to reference this with a data block too. AMI IDs are globally unique identifiers so I think it's acceptable to reference it somewhere.